app/assets/javascripts/components/form/teacher_form.es6.jsx

```diff
@@ -159,6 +159,71 @@ class TeacherForm extends React.Component {
     });
   }
 
+  handleAddressChange(event) {
+    var componentForm = {
+      street_number: 'short_name',
+      route: 'long_name',
+      locality: 'long_name',
+      administrative_area_level_1: 'short_name',
+      postal_code: 'short_name'
+    };
+
+    var autocomplete = new google.maps.places.Autocomplete(document.getElementById("address"));
+    this.geolocate(autocomplete);
+    autocomplete.addListener("place_changed", function() {
+      var place = autocomplete.getPlace();
+
+      // Clear any old address information
+      for (var i = 2; i < componentForm.length; i++) {
+        document.getElementById(componentForm[i]).value = '';
+        document.getElementById(componentForm[i]).disabled = false;
+      }
+      document.getElementById("address").value = '';
+      document.getElementById("address").disabled = false;
+
+      // Get each component of the address from the place details
+      // and fill the corresponding field on the form.
+      var street_number, street_name;
+      for (var i = 0; i < place.address_components.length; i++) {
+        var addressType = place.address_components[i].types[0];
+        if (componentForm[addressType]) {
+          var val = place.address_components[i][componentForm[addressType]];
+          if (addressType == "administrative_area_level_1") {
+            val = STATES.indexOf(val);
+            document.getElementById(addressType).value = val;          
+          } else if (addressType == "street_number") {
+            street_number = val;
+          } else if (addressType == "route") {
+            street_name = val;
+          } else {
+            document.getElementById(addressType).value = val;          
+          }
+        }
+      }
+      if (street_number != null && street_name != null) {
+        document.getElementById("address").value = street_number + " " + street_name;
```

Comment:
changing the text directly as the user is interacting with it may get them confused and frustrated. a good alternative to this would be to allocate a dropdown of suggestions, and if a user chooses it then fill the text of the form with id `address` with the text they selected. so, as they type, this function would get triggered and repopulate the dropdown with its suggestions. this is gonna involve probably making a custom component for this form specifically since you also need to handle logic for displaying the dropdown and also populating the input field if a user decides to choose a suggestion, but this will be better and feel more natural to users in the long run. you could also just move this method into the new component as well, since it seems like only the `address` element needs this method. you can also just use the same component for both the student and teacher form